src/ui/interface.js

```diff
@@ -858,28 +859,49 @@ var UI = class UI {
 	 *
 	 * Show the dash and hide some buttons
 	 */
-	toggleDash() {
-		let game = this.game;
 
+	toggleDash: function (randomize) {
+    let game = this.game;
 		if (!this.$dash.hasClass("active")) {
-			this.showCreature(game.activeCreature.type, game.activeCreature.team);
+			if (randomize) {
+				const activePlayer = game.players[game.activeCreature.player.id];
+				const deadOrSummonedTypes = activePlayer.creatures.map(creature => creature.type)
+				const availableTypes = activePlayer.availableCreatures.filter(el => !deadOrSummonedTypes.includes(el))
+				// Optional: randomize array to grab a new creature every toggle
+				for (let i = availableTypes.length - 1; i > 0; i--) {
+					let j = Math.floor(Math.random() * (i + 1));
+					let temp = availableTypes[i];
+					availableTypes[i] = availableTypes[j];
+					availableTypes[j] = temp;
+				}
+				// Grab the first creature we can afford (if none, default to priest)
+				let typeToPass = "--";
+				availableTypes.some(creature => {
+					const lvl = creature.substring(1, 2) - 0;
+					const size = game.retreiveCreatureStats(creature).size - 0;
+					const plasmaCost = lvl + size;
+					return plasmaCost <= activePlayer.plasma ? ((typeToPass = creature), true) : false;
+				});
+				this.showCreature(typeToPass, game.activeCreature.team);
+			} else {
+				this.showCreature(game.activeCreature.type, game.activeCreature.team);
+			}
 		} else {
 			this.closeDash();
 		}
-	}
+	},
+
 
 	closeDash() {
 		let game = this.game;
-
```

Comment:
Generally all the code here has the variable definitions separated for easier distinction between variables and code block at the top of functions.